src/preprocessing/passes/non_clausal_simp_sat.cpp

```diff
@@ -0,0 +1,483 @@
+/*********************                                                        */
+/*! \file non_clausal_simp.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Haoze Wu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2018 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Non-clausal simplification preprocessing pass.
+ **
+ ** Run the nonclausal solver and try to solve all assigned theory literals.
+ **/
+
+ #include "preprocessing/passes/non_clausal_simp_sat.h"
+ #include <cryptominisat5/cryptominisat.h>
+
+ #include "prop/registrar.h"
+ #include "prop/cnf_stream.h"
+ #include "prop/sat_solver_factory.h"
+
+ #include "proof/proof_manager.h"
+
+ #include "context/cdo.h"
+ #include "options/proof_options.h"
+ #include "smt/smt_statistics_registry.h"
+ #include "theory/theory_model.h"
+#include <ctime>
```

Comment:
Please follow the [styleguide for includes](https://github.com/CVC4/CVC4/wiki/Code-Style-Guidelines#names-and-order-of-includes) (also applies to the other changes in the includes). Also, what is `<ctime>` used for?